### DISCUSSION TITLE
Build error trying to build Maui Source

### DISCUSSION THREAD

**Discussion by nogginbox** [dotnet/maui#6633] [Q&A] [answered]
I'm trying to build the latest main branch version of Maui so I can write some unit tests for my [bug fix PR](/dotnet/maui/pull/6085).

However I'm getting several errors. I think this is the main one and the rest should/may go away if I can fix it:

```
Error CS1705 Assembly 'Microsoft.Maui.Graphics' with identity 'Microsoft.Maui.Graphics, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null' uses
'Microsoft.MacCatalyst, Version=15.4.300.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'
which has a higher version than referenced assembly 'Microsoft.MacCatalyst' with identity
'Microsoft.MacCatalyst, Version=15.4.200.0, Culture=neutral, PublicKeyToken=84e04ff9cfb79065'
Essentials (netstandard2.1)
```

I'm building using the latest preview of VS 2012. Is there anything I'm missing?

**Comment by pictos** [comment, marked as answer]
@NogginBox can you try to run?

```
dotnet tool restore
dotnet cake --target=VS-DOGFOOD
```

At the end the command should launch your VS

**Reply by nogginbox** [reply]
That took a while to run. I am now getting different errors that comes up when loading the project. I get lots that are similar to this:
```
The project 'Microsoft.Maui.Dependencies' ran into a problem during the last operation:
The value of the 'TargetFrameworks' property is inconsistent between target frameworks.
This property must be identical for NuGet restore to function correctly.
The value 'net6.0;net6.0-ios;net6.0-maccatalyst;net6.0-android;net6.0-windows10.0.19041;net6.0-windows10.0.20348'
from the 'Debug|AnyCPU|net6.0' configuration will be used, other target frameworks may fail to pick NuGet assets.
You may need to reload the solution after fixing the problem. 
```

Then when building I get a load of type or namespace x does not exist messages. I'm not sure I like this dogfood.

**Reply by pictos** [reply]
Did you install your workloads using the VS Installer? Are they in the latest version?

**Reply by pictos** [reply]
If that doesn't work, you can try this another target to build and use a local version of net6
https://github.com/dotnet/maui/blob/main/.github/DEVELOPMENT.md#compile-using-a-local-bindotnet

**Reply by nogginbox** [reply]
I got confused by the instruction on the [Maui Development Guide](/dotnet/maui/blob/main/.github/DEVELOPMENT.md) that said:

> In most cases, when you have Visual Studio installed with the .NET workloads checked, these steps are not required.

I thought it meant all the steps, but maybe it's just referring to the first section.

I thought the workloads would have been installed with VS2022 preview installer, but maybe they're not the very latest. So I'm trying to install the workloads with the command line, but I'm getting this message:
```
C:\Users\Nogginbox\Code\Maui-Src>dotnet workload install maui
Skipping NuGet package signature verification.
Workload installation failed. Rolling back installed packs...
Workload installation failed: Failed to verify authenticode signature for C:\ProgramData\dotnet\workloads\microsoft.net.workload.mono.toolchain.manifest-6.0.300.msi.x64\6.0.5\Microsoft.NET.Workload.Mono.ToolChain.Manifest-6.0.300.6.0.5-x64.msi.
Error: 0x800b0100. No signature was present in the subject.
```

P.S. Thanks for your help.

**Reply by pictos** [reply]
@NogginBox are you running the terminal with admin privileges?

P.S.: You're welcome

**Reply by mattleibow** [reply]
Ah the signing... Maybe install a later version of the 6.0.300 sdk from the github.com/dotnet/installer repo to get the unsigned bits

**Reply by nogginbox** [reply]
I've tried a uninstall vs and everything and start again, but not having much luck.

After installing latest dotnet sdk, I'm getting the following error when installing maui workload:

> Workload installation failed: One or more errors occurred. (microsoft.maui.core.ref.android.msi.x64::6.0.300-rc.3.5620 is not found in NuGet feeds https://api.nuget.org/v3/index.json;C:\Program Files (x86)\Microsoft SDKs\NuGetPackages\".)

Is there a secret nuget feed I need to hook up to?

**Reply by nogginbox** [reply]
I found some feed sources to add that got me past that error:

`dotnet workload install maui-android --source "https://api.nuget.org/v3/index.json" --source "https://pkgs.dev.azure.com/dnceng/public/_packaging/6.0.100-preview.7.21379.14-shipping-1/nuget/v3/index.json" --source "https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json" --source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json"`

**Are these feeds okay?** I should have checked them first, but was getting desperate.

**Reply by nogginbox** [reply]
Think I'm really close now. I'm just getting one error:

```
Error occurred while restoring NuGet packages: Invalid restore input. The original target frameworks value must match the aliases.
Original target frameworks: net6.0;net6.0-ios;net6.0-maccatalyst;net6.0-android;net6.0-windows10.0.19041;net6.0-windows10.0.20348;net6.0-tizen,
                   aliases: net6.0;net6.0-ios;net6.0-maccatalyst;net6.0-android;net6.0-windows10.0.19041;net6.0-windows10.0.20348.
Input files: C:\Users\Nogginbox\Code\Maui\Maui-Src\src\BlazorWebView\src\Maui\Microsoft.AspNetCore.Components.WebView.Maui.csproj.
```

The difference is that original target frameworks has **net6.0-tizen** in it, and aliases does not.

**Reply by pictos** [reply]
@NogginBox just to be sure, your local repo is synced with the `main` branch of the .NET MAUI repo on GitHub?

**Reply by nogginbox** [reply]
@pictos Yep. I'm all caught up with `main`.

**Reply by nogginbox** [reply]
**Got it working!!!** Noticed a commit to main with updated instructions for development.md that had updated sources in it.

I'm not sure if that is what finally sorted it as I've tried so many things, but it now compiles and my unit tests run.

**Reply by pictos** [reply]
Glad that you are able to make it work! Can't wait for your PR!

**Comment by nogginbox** [comment]
**The short answer is make sure you read the setup instructions on the [Maui Development Guide](/dotnet/maui/blob/main/.github/DEVELOPMENT.md) carefully and follow them all.**

In particular make sure you specify the --source parameters when installing the workloads as this seems to get you the extra in progress stuff you need.
